src/config/mongoose_config_parser_toml.erl

```diff
@@ -471,30 +472,700 @@ process_module([Mod|_] = Path, Opts) ->
+module_opt([<<"api_version">>, <<"mod_push_service_mongoosepush">>|_], V) ->
+    [{api_version, b2l(V)}];
+module_opt([<<"max_http_connections">>, <<"mod_push_service_mongoosepush">>|_], V) ->
+    [{max_http_connections, V}];
+module_opt([<<"access">>, <<"mod_register">>|_], V) ->
+    [{access, b2a(V)}];
 module_opt([<<"welcome_message">>, <<"mod_register">>|_], V) ->
     [{welcome_message, {b2l(V)}}];
+module_opt([<<"registration_watchers">>, <<"mod_register">>|_], V) ->
+    [{registration_watchers, V}];
+module_opt([<<"password_strength">>, <<"mod_register">>|_], V) ->
+    [{password_strength, V}];
 module_opt([<<"ip_access">>, <<"mod_register">>|_] = Path, V) ->
     Rules = parse_list(Path, V),
     [{ip_access, Rules}];
-module_opt([<<"access">>, <<"mod_register">>|_], V) ->
-    [{access, b2a(V)}];
+module_opt([<<"routes">>, <<"mod_revproxy">>|_] = Path, V) ->
+    Routes = parse_list(Path, V),
+    [{routes, Routes}];
+module_opt([<<"versioning">>, <<"mod_roster">>|_], V) ->
+    [{versioning, V}];
+module_opt([<<"store_current_id">>, <<"mod_roster">>|_], V) ->
+    [{store_current_id, V}];
+module_opt([<<"ldap_groupattr">>, <<"mod_shared_roster_ldap">>|_], V) ->
+    [{ldap_groupattr, b2l(V)}];
+module_opt([<<"ldap_groupdesc">>, <<"mod_shared_roster_ldap">>|_], V) ->
+    [{ldap_groupdesc, b2l(V)}];
+module_opt([<<"ldap_userdesc">>, <<"mod_shared_roster_ldap">>|_], V) ->
+    [{ldap_userdesc, b2l(V)}];
+module_opt([<<"ldap_userid">>, <<"mod_shared_roster_ldap">>|_], V) ->
+    [{ldap_userid, b2l(V)}];
+module_opt([<<"ldap_memberattr">>, <<"mod_shared_roster_ldap">>|_], V) ->
+    [{ldap_memberattr, b2l(V)}];
+module_opt([<<"ldap_memberattr_format">>, <<"mod_shared_roster_ldap">>|_], V) ->
+    [{ldap_memberattr_format, b2l(V)}];
+module_opt([<<"ldap_memberattr_format_re">>, <<"mod_shared_roster_ldap">>|_], V) ->
+    [{ldap_memberattr_format_re, b2l(V)}];
+module_opt([<<"ldap_auth_check">>, <<"mod_shared_roster_ldap">>|_], V) ->
+    [{ldap_auth_check, V}];
+module_opt([<<"ldap_user_cache_validity">>, <<"mod_shared_roster_ldap">>|_], V) ->
+    [{ldap_user_cache_validity, V}];
+module_opt([<<"ldap_group_cache_validity">>, <<"mod_shared_roster_ldap">>|_], V) ->
+    [{ldap_group_cache_validity, V}];
+module_opt([<<"ldap_user_cache_size">>, <<"mod_shared_roster_ldap">>|_], V) ->
+    [{ldap_user_cache_size, V}];
+module_opt([<<"ldap_group_cache_size">>, <<"mod_shared_roster_ldap">>|_], V) ->
+    [{ldap_group_cache_size, V}];
+module_opt([<<"ldap_rfilter">>, <<"mod_shared_roster_ldap">>|_], V) ->
+    [{ldap_rfilter, b2l(V)}];
+module_opt([<<"ldap_gfilter">>, <<"mod_shared_roster_ldap">>|_], V) ->
+    [{ldap_gfilter, b2l(V)}];
+module_opt([<<"ldap_ufilter">>, <<"mod_shared_roster_ldap">>|_], V) ->
+    [{ldap_ufilter, b2l(V)}];
+module_opt([<<"buffer_max">>, <<"mod_stream_management">>|_], V) ->
+    [{buffer_max, V}];
+module_opt([<<"ack_freq">>, <<"mod_stream_management">>|_], V) ->
+    [{ack_freq, V}];
```

Comment:
There are atoms like `infinity` or `no_buffer` accepted in the code. I am not sure if we need to support them if they are undocumented. Could you double check it?